sw/device/tests/aes_idle.c

```diff
@@ -0,0 +1,229 @@
+  // Toggle the hint twice so that it ends up in its original state.
+  for (int j = 0; j < 2; ++j) {
+    bool expected = !enabled;
+    CHECK(dif_clkmgr_hintable_clock_set_hint(
+              clkmgr, kTopEarlgreyHintableClocksMainAes,
+              expected ? kDifClkmgrToggleEnabled
+                       : kDifClkmgrToggleDisabled) == kDifClkmgrOk);
+    CHECK(dif_clkmgr_hintable_clock_get_hint(clkmgr, kTopEarlgreyHintableClocksMainAes, &enabled) ==
+          kDifClkmgrOk);
+    CHECK(enabled == expected);
+
+    // If the clock hint is enabled then the clock should always be enabled.
+    if (enabled) {
+      bool status = false;
+      CHECK(dif_clkmgr_hintable_clock_get_enabled(clkmgr, kTopEarlgreyHintableClocksMainAes, &status) ==
+            kDifClkmgrOk);
+      CHECK(status, "clock %u hint is enabled but status is disabled", kTopEarlgreyHintableClocksMainAes);
+    }
+  }
+}
+
+
+bool test_main(void) {
+  const dif_clkmgr_params_t params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR),
+      .last_gateable_clock = kTopEarlgreyGateableClocksLast,
+      .last_hintable_clock = kTopEarlgreyHintableClocksLast,
+  };
+
+  dif_aes_t aes;
+  dif_clkmgr_t clkmgr;
+
+  CHECK(dif_clkmgr_init(params, &clkmgr) == kDifClkmgrOk);
+
+  LOG_INFO("Entropy complex ini...");
+  // First of all, we need to get the entropy complex up and running.
+  entropy_testutils_boot_mode_init();
+
+  LOG_INFO("AES ini...");
+  // Initialise AES.
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+  CHECK_DIF_OK(dif_aes_reset(&aes));
+
+  LOG_INFO("Set AES hint");
+  set_aes_hint(&clkmgr);
+
+  // Mask the key. Note that this should not be done manually. Software is
+  // expected to get the key in two shares right from the beginning.
+  uint8_t key_share0[KEY_LENGTH_IN_BYTES];
+  for (int i = 0; i < KEY_LENGTH_IN_BYTES; ++i) {
+    key_share0[i] = kKey[i] ^ kKeyShare1[i];
+  }
+
+  LOG_INFO("copy keyshare to AES...");
+  // "Convert" key share byte arrays to `dif_aes_key_share_t`.
+  dif_aes_key_share_t key;
+  memcpy(&key.share0[0], &key_share0[0], KEY_LENGTH_IN_BYTES);
+  memcpy(&key.share1[0], &kKeyShare1[0], KEY_LENGTH_IN_BYTES);
+
+  LOG_INFO("triggeringECB encryption...");
+  // Setup ECB encryption transaction.
+  dif_aes_transaction_t transaction = {
+      .key_len = kDifAesKey256,
+      .mode = kDifAesModeEncrypt,
+      .operation = kDifAesOperationAuto,
+  };
+  CHECK_DIF_OK(dif_aes_start_ecb(&aes, &transaction, key));
+
+  // "Convert" plain data byte arrays to `dif_aes_data_t`.
+  dif_aes_data_t in_data_plain;
+  memcpy(&in_data_plain.data[0], &kPlainText[0], TEXT_LENGTH_IN_BYTES);
+
+
+  LOG_INFO("polling for aes input ready...");
+  // Load the plain text to trigger the encryption operation.
+  while (!aes_input_ready(&aes)) {
+  }
+  CHECK_DIF_OK(dif_aes_load_data(&aes, in_data_plain));
+
+  //Verify that the AES clk hint status within clkmgr now reads 1 (AES is not
```

Comment:
Disable the AES clock hint to 0 and verify the hist status is 1 because clkmgr cannot disable AES clock since it just started an operation.

Reply:
By the time the difs can set this, it's no longer active. https://screenshot.googleplex.com/3tJRKqnGujekHoW